File: ruby/strategy/DuckSimulator/client/mallard_duck.rb
require "./client/duck"
require './behavior/fly_behavior'
require './behavior/quack_behavior'

class MallardDuck < Duck

  def initialize(duck_name)
    super(duck_name)
    @fly_behavior = FlyWithWings.new(duck_name)
    @quack_behavior = Quack.new(duck_name)
  end

end

File: javascript/strategy/DuckSimulator/client/MallardDuck.js
import {Duck} from "./Duck.js";
import {FlyWithWings} from '../behavior/FlyBehavior.js'
import {Quack} from '../behavior/QuackBehavior.js'

export class MallardDuck extends Duck {

    constructor(duck_name) {
        super(duck_name)
        this.fly_behavior = new FlyWithWings(duck_name)
        this.quack_behavior = new Quack(duck_name)
    }
}